### ISSUE TITLE
Cross Platform FFI ultra fast method spoted

### ISSUE THREAD

**Issue by lemanschik** [just-js/lo#34]
i saw on your todo cross platform ffi read this repo about that lets apply same method.

- https://github.com/bengl/sbffi

**Comment by xyloflake** [comment]
> i saw on your todo cross platform ffi read this repo about that lets apply same method.
> 
> * https://github.com/bengl/sbffi
> 
> [dyncall](https://dyncall.org/) is used to make dynamic calls to native functions. In order to avoid some cost of translating JavaScript values into raw C types, a shared buffer is used for both arguments and return values. Writing values to a buffer turns out to be quite a bit faster than unpacking them in native code.

FYI, lo's ffi is very fast. SBFFI is one of the slowest ones when it's compared. This is a comparison as posted on the discord server by @billywhizz 
![image](https://github.com/just-js/lo/assets/142194553/4d1c3225-4e46-4ae7-897e-887bc5796043)

**Comment by billywhizz** [comment]
> FYI, lo's ffi is very fast. SBFFI is one of the slowest ones when it's compared. This is a comparison as posted on the discord server by @billywhizz 

yes. there is quite some overhead. that graph shows a call to a function that is doing quite some work (counting lines in a buffer). i'll see if i can get one that just tests the ffi overhead on a noop(). among the mainstream runtimes, deno ffi and then bindings seem to be quite a bit less overhead than any of the other ffi methods. afaik, deno ffi and lo ffi/bindings are only ones currently using v8 fastcalls.

**Comment by xyloflake** [comment]
@billywhizz see if you can write a blog post about how bindings in lo work, pretty darn sure everyone would love it. 

Also why is this issue still open? Will you be using to track cross platform ffi support? In that case please change the title of the issue.

**Comment by lemanschik** [comment]
@xyloflake whats your take about #40
